src/core/utils/helpers/url.ts

```diff
@@ -123,3 +123,32 @@ export const removeQueryParametersFromUrl = (url: URL, parameter: string) => {
   url.searchParams.delete(parameter);
   return url;
 };
+
+type RedirectToLoginAndBackParams = {
+  authUrl: URL;
+  returnUrl: URL;
+  trackingFunction?: () => Promise<unknown>;
+};
+export function redirectToLoginAndBack({
+  authUrl,
+  returnUrl,
+  trackingFunction
+}: RedirectToLoginAndBackParams) {
+  // If we are in storybook context just redirect to the login story.
+  if (authUrl.pathname === "/iframe.html" && window.top) {
+    const { origin } = new URL(getCurrentLocation());
+    const storybookRedirect = `${origin}/?path=/story/sb-utilities-adgangsplatformen--sign-in`;
+    // We don't use redirectTo() because that would redirect inside the storybook iframe.
+    window.top.location.href = storybookRedirect;
+    return;
+  }
+  const { pathname, search } = returnUrl;
+  const localPathToReturnTo = `${pathname}${search}`;
+  const redirectUrl = appendQueryParametersToUrl(authUrl, {
+    "current-path": localPathToReturnTo
+  });
```

Comment:
We could create a global url function called eg.: `createAuthenticationRedirectUrl`

Reply:
makes total sense! Roger 😉